### PR TITLE
Feat: user find, findOne 구현 및 기존 function 대체 #121

### PR DESCRIPTION
**Pull request by phjppo0918**

### CODE REVIEW

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -4,21 +4,44 @@ module.exports = {
...
+    searchDetail: async function(req, res) {
+
+        const projection = {
+            _id: true,serviceNumber:true, name: true, rank: true, title:true,
+            status: true, group: true, email: true, tel: true, lastLogin: true,
+            lastLogin: true, firseLogin:true, bookmarks: true, subscriptions: true
+        };
+        try {
+            const result = await userService.findOne({_id: req.params.id}, projection);
+            res.status(201).send(result);   // 201 Created
```
Comment (134130) [inline backend/controllers/userController.js:39]: Status code `201` is using for when resource has created.
Status code `200` is cut out for this function.

File: backend/controllers/userController.js [hunk 2]
```diff
@@ -4,21 +4,44 @@ module.exports = {
...
+        const projection = {
+            _id: true, name: true, rank: true, status: true,
+            group: true, email: true, tel: true, lastLogin: true
         };
 
         try {
-            const result = await userService.search(query, projection);
+            const result = await userService.find(req.query, projection);
             res.status(200).send(result);
         } catch(err) {
```
Comment (134130) [inline backend/controllers/userController.js:25]:
```javascript
res.status(err.status).send(err.message);
```
this code can occur another error when err.status had not defined.

```javascript
res.status(err.status || 500).send(err.message);
```
this code can send `500` when err.status was not defined.

File: backend/controllers/userController.js [hunk 3]
```diff
@@ -4,21 +4,44 @@ module.exports = {
...
 
+    searchDetail: async function(req, res) {
+
+        const projection = {
+            _id: true,serviceNumber:true, name: true, rank: true, title:true,
+            status: true, group: true, email: true, tel: true, lastLogin: true,
+            lastLogin: true, firseLogin:true, bookmarks: true, subscriptions: true
+        };
+        try {
+            const result = await userService.findOne({_id: req.params.id}, projection);
```
Comment (134130) [inline backend/controllers/userController.js:38]: `404 NotFound Error` is unhandled.

**Review by 134130 — CHANGES_REQUESTED** [review]

File: backend/controllers/userController.js (hunk 1 above)
Comment (ntcho) [reply backend/controllers/userController.js:39]: As @134130 said, `200` should be used in this case.

**Review by ntcho — APPROVED** [review]
There can be more comments explaining what each function does, but approved for now.

File: backend/controllers/userController.js (hunk 3 above)
Comment (phjppo0918) [reply backend/controllers/userController.js:38]:
```
findOne: async function(params, projection = {}) {

		const result = await User
			.findOne(params, projection)
			.catch(err => {
				throw new RuntimeError(err.message);
			});

		if(result === null) throw new NotFoundError('Not Found: 검색 결과가 없습니다.');

		return result;
	},
```
 404 errors are handled in userService.

**Pull request merged**